### ISSUE TITLE
[Question] How to setup on VSCode to be able jump to definition

### ISSUE THREAD

**Issue by bombi0104** [thecodingmachine/react-native-boilerplate#180]
import fetchOneUserService from '@/Services/User/FetchOne'

When import like this, i can't touch CMD + click on the `fetchOneUserService` to jump to the definition.
What problem with my VSCode environment ? Can you advise me to fix it out.

Thank you

**Comment by JeremyDolle** [comment]
Hi @bombi0104, can you touch CMD + click on `'@/Services/User/FetchOne'` ?

**Comment by pqminh** [comment]
I can't see any suggestion for useTheme, Images or Layout...

**Comment by JeremyDolle** [comment]
Hi @pqminh, do you are able to jump into definition? 

What do you expecting? Have an autocomplete for importing the `useTheme` hook or autocomplete in the restructuring to get Layout, Gutters... ?

Autocomplete suggestions is maybe another issue 🙂 Let me know and I will make some research for that !

**Comment by pqminh** [comment]
Jsconfig is working fine in vscode
But with useTheme hook, so difficult to see any props to be defined.

**Comment by JeremyDolle** [comment]
Yes, it is because the style is generated and therefore hidden behind generator functions. With this implementation, you have the ability to add, modify, delete classes and be able to create multiple overloading themes in cascade. If I try to create an autocomplete theme, I lose flexibility and dynamics. I add in my roadmap to find a possible solution for that 😉

**Comment by dthomason** [comment]
I fixed this by pulling the StyleSheet.create portions out above the export default.  Gave it a const then pulled the type from that.
`/Layout.ts`
```
...//
const styles = StyleSheet.create({...//
...//
type LayoutOptions = typeof styles;
```
then specified LayoutOptions as the return type as well as modifying `theme.types.ts` 
```
export type ThemeLayout = Partial<LayoutOptions>;
```

**Comment by JeremyDolle** [comment]
@dthomason are you interesting about making a pull request with your fix ?

**Comment by mediarl** [comment]
Hi,

If someone reads it, I had the same issue with jumping to the definition. Just add "baseUrl": "." 
And it should fix it.

Great project by the way, thanks !

```
{
    "compilerOptions": {
        "baseUrl": ".",
        "paths": {
            "@/*": [
                "./src/*"
            ]
        }
    }
}
```

**Comment by pqminh** [comment]
@JeremyDolle 
Do you think about this solution(JS)?
<img width="1126" alt="Screen Shot 2022-01-06 at 11 01 29" src="https://user-images.githubusercontent.com/13993984/148326378-75390b5c-c108-4106-a7ef-862f7610f715.png">
<img width="750" alt="Screen Shot 2022-01-06 at 11 01 48" src="https://user-images.githubusercontent.com/13993984/148326386-15befeae-478f-47fe-b742-dd8f732a5e26.png">
